src/wovnio/wovnphp/Url.php

```diff
@@ -174,8 +183,45 @@ public static function removeLangCode($uri, $pattern, $lang_code, $lang_param_na
             case 'path':
             default:
                 // limit to one replacement
-                return preg_replace('/\/' . $lang_code . '(\/|$)/i', '/', $uri, 1);
+                $prefix = empty($site_prefix_path) ? '' : '/' . $site_prefix_path;
+                return preg_replace("@$prefix/$lang_code(/|$)@i", "$prefix/", $uri, 1);
                 break;
         }
     }
+
+    public static function shouldIgnoreBySitePrefixPath($uri, $settings)
+    {
+        if ($settings['site_prefix_path'] &&
+            $settings['url_pattern_name'] &&
+            $settings['url_pattern_name'] === 'path'
+        ) {
```

Comment:
You said the store initializes default settings, so we should be able to remove 
```
$settings['site_prefix_path'] &&
            $settings['url_pattern_name'] &&
```
right?

Reply:
`$settings['site_prefix_path']` could be null after initlization.
I will remove `$settings['url_pattern_name']`.

Reply:
Done